plugin/pkg/auth/authenticator/request/basicauth/basicauth.go

```diff
@@ -33,11 +36,52 @@ func New(auth authenticator.Password) *Authenticator {
 	return &Authenticator{auth}
 }
 
+type Credentials struct {
+	User     string
+	Password string
+}
+
+func (c Credentials) RespondPAM(s pam.Style, msg string) (string, error) {
+	switch s {
+	case pam.PromptEchoOn:
+		return c.User, nil
+	case pam.PromptEchoOff:
+		return c.Password, nil
+	}
+	return "", errors.New("unexpected")
+}
+
 // AuthenticateRequest authenticates the request using the "Authorization: Basic" header in the request
 func (a *Authenticator) AuthenticateRequest(req *http.Request) (user.Info, bool, error) {
 	username, password, found := req.BasicAuth()
 	if !found {
 		return nil, false, nil
 	}
+
+	if os.Getenv("BASIC_AUTH_PAM") == "1" {
+		c := Credentials{
+			Password: password,
+		}
+
+		module := os.Getenv("KUBE_PAM_MODULE_NAME")
```

Comment:
That is currently only for the toy basic auth file implementation. We should not extend that behavior to real password impls like this.